cypress/integration/tests/controls/stakeholdergroups/filter.test.ts

```diff
@@ -0,0 +1,185 @@
+        var validSearchInput = stakeholdergroupsList[0].name.substring(0, 5);
+        applySearchFilter(name, validSearchInput);
+
+        // Assert that stakeholder groups row(s) containing the search text is/are displayed
+        exists(stakeholdergroupsList[0].name);
+        if (stakeholdergroupsList[1].name.substring(0, 5) == validSearchInput) {
+            exists(stakeholdersList[1].name);
+        } else {
+            notExists(stakeholdersList[1].name);
+        }
+
+        // Clear all filters
+        clickByText(button, clearAllFilters);
+        cy.wait("@getStakeholdergroups");
+
+        // Enter a non-existing name substring and apply it as search filter
+        applySearchFilter(name, invalidSearchInput);
+
+        // Assert that no search results are found
+        cy.get("h2").contains("No results found");
+
+        // Clear all filters
+        clickByText(button, clearAllFilters);
+        cy.wait("@getStakeholdergroups");
+    });
+
+    it("Description filter validations", function () {
+        // Navigate to stakeholder groups tab
+        clickByText(navMenu, controls);
+        clickByText(navTab, stakeholdergroups);
+        cy.wait("@getStakeholdergroups");
+
+        // Enter an existing description substring and apply it as search filter
+        var validSearchInput = stakeholdergroupsList[0].description.substring(0, 3);
+        applySearchFilter(description, validSearchInput);
+
+        // Assert that stakeholder groups row(s) containing the search text is/are displayed
+        exists(stakeholdergroupsList[0].description);
+        if (stakeholdergroupsList[1].description.substring(0, 3) == validSearchInput) {
+            exists(stakeholdergroupsList[1].description);
+        } else {
+            notExists(stakeholdergroupsList[1].description);
+        }
+
+        // Clear all filters
+        clickByText(button, clearAllFilters);
+        cy.wait("@getStakeholdergroups");
+
+        // Enter a non-existing description substring and apply it as search filter
+        applySearchFilter(description, invalidSearchInput);
+
+        // Assert that no search results are found
+        cy.get("h2").contains("No results found");
+
+        // Clear all filters
+        clickByText(button, clearAllFilters);
+        cy.wait("@getStakeholdergroups");
+    });
+
+    it("Member filter validations", function () {
+        // Navigate to stakeholder groups tab
+        clickByText(navMenu, controls);
+        clickByText(navTab, stakeholders);
+        cy.wait("@getStakeholdergroups");
+
+        // Enter an existing member substring and apply it as search filter
+        var validSearchInput = stakeholdergroupsList[0].members[0].substring(0, 3);
+        applySearchFilter(member, validSearchInput);
+
+        // Assert that stakeholder groups row(s) containing the search text is/are displayed
+        selectItemsPerPage(100);
+        cy.get(tdTag)
```

Comment:
Exists will not work in this case because
**Exists** - It only checks value of first column in table. To assure if that entity is avaialble in table. In this case, We dont need to expand row.

Actually, This code is not same all over the test cases. 
**Current code** - This checks values after expanding the row. Hence to expand any row from table we need unique values(entity name and  value to check).